queue/queue.py

```diff
@@ -0,0 +1,54 @@
+# _*_ coding: utf-8- _*_
+from __future__ import unicode_literals
+
+
+class Node(object):
+    def __init__(self, val=None, next_node=None):
+        self.val = val
+        self.next_node = next_node
+
+
+class Queue(object):
+    def __init__(self, back=None, front=None):
+        self.back = back
+        self.front = front
+        self.size_ = 0
+
+    def enqueue(self, val):
+        """
+        Insert new Node at the back of the queue. If queue is empty, first Node
+        becomes Front of queue. Otherwise, new Nodes are added to back of
+        queue.
+            arg:
+                val: Value associated as the data of the Node
+        """
+        if self.front is None:
+            node = Node(val)
+            self.front = node
+            self.back = node
```

Comment:
You can do self.front = self.back = node
But I can't really say its better.. Sometimes it gets hard to read when its all in one line. But its a good trick to have in your pocket.